### PR TITLE
Fix ping not return pong

### PR DESCRIPTION
**Pull request by zhongwencool**
1. `./bin/emqx ping` should pong on success.
2. Change HTTP API's max limit from 100 to 1000.
3. fix authn api return 500 when config format error.

### CODE REVIEW

File: bin/emqx [hunk 1]
```diff
@@ -587,6 +587,7 @@ case "${COMMAND}" in
 
     ping)
         assert_node_alive
+        echo pong
```
Comment (zmstone) [inline bin/emqx:590]: why ?

Comment (zhongwencool) [reply bin/emqx:590]:
```shell
>   ./bin/emqx ping
Node 'emqx@127.0.0.1' not responding to pings.
ERROR: node_is_not_running!
>       ./bin/emqx ping
pong
````
pong to remind the user is ok.

File: apps/emqx_authn/src/emqx_authn_api.erl [hunk 2]
```diff
@@ -919,6 +920,14 @@ serialize_error({unknown_authn_type, Type}) ->
 serialize_error({bad_authenticator_config, Reason}) ->
     {400, #{code => <<"BAD_REQUEST">>,
             message => binfmt("Bad authenticator config ~p", [Reason])}};
+serialize_error(#{reason := {Mod, ValidationErrors}, type := Type} = Error) ->
+    ?SLOG(error, Error),
```
Comment (savonarola) [inline apps/emqx_authn/src/emqx_authn_api.erl:924]: Do we intentionally log only this concrete error?

Comment (zhongwencool) [reply apps/emqx_authn/src/emqx_authn_api.erl:924]: sorry for this error, I will remove this, add a test case to cover this.

**Comment by savonarola** [comment]
@zhongwencool  Could you please describe abit which config update brought 500 error?

I am wondering why this test https://github.com/emqx/emqx/blob/master/apps/emqx_authn/test/emqx_authn_api_SUITE.erl#L137 do not catch that.

**Comment by zhongwencool** [comment]
> @zhongwencool Could you please describe a bit which config update brought 500 error?
> 
> I am wondering why this test https://github.com/emqx/emqx/blob/master/apps/emqx_authn/test/emqx_authn_api_SUITE.erl#L137 do not catch that.

image

**Comment by savonarola** [comment]
Thanks! I'll just see how we reproduce that in tests

**Comment by savonarola** [comment]
I got why we have this error.

When we compose authenticator type for validation we make union of `ref(Module, config)` types and lose `Module`'s own validations. So the error is occures not in minirest filter, but when we actually make an update request.

**Comment by savonarola** [comment]
Reproducing in tests is easy: in `test_authenticator(s)` we add
```erlang
    InvalidConfig1 = ValidConfig#{method => <<"get">>,
                                  headers => #{<<"content-type">> => <<"application/json">>}},
    {ok, 400, _} = request(
                     post,
                     uri(PathPrefix ++ ["authentication"]),
                     InvalidConfig1),
```

**Review by zmstone — DISMISSED** [review]

**Comment by savonarola** [comment]
@zhongwencool made a PR where I added added tests covering this: https://github.com/emqx/emqx/pull/6288 (and proposed the actual fix)

**Review by savonarola — APPROVED** [review]

**Comment by zhongwencool** [comment]
Previously, using ct:sleep/1  to sync logs to disk, there may be a situation where sleep time is too short and not yet synced.
Now force to use logger_disk_log_h:filesync/1 to synchronize.

```erlang
021-11-24T20:07:50.1661865Z %%% emqx_trace_handler_SUITE ==> t_trace_topic: FAILED
2021-11-24T20:07:50.1662307Z %%% emqx_trace_handler_SUITE ==> 
2021-11-24T20:07:50.1662807Z Failure/Error: ?assertNotEqual(nomatch, binary : match ( Bin , [ << "hi1" >> ] ))
2021-11-24T20:07:50.1663326Z   expected not: == nomatch
2021-11-24T20:07:50.1663674Z            got:    nomatch
2021-11-24T20:07:50.1663956Z           line: 134

```

**Review by zmstone — APPROVED** [review]

**Pull request merged**